core/chains/evm/headtracker/head_listener.go

```diff
@@ -28,32 +32,62 @@ var (
 	}, []string{"evmChainID"})
 )
 
-type headListener struct {
-	config           Config
-	ethClient        evmclient.Client
+type headListener[
+	H commontypes.Head[BLOCK_HASH],
+	HTH htrktypes.Head[H, BLOCK_HASH, ID],
```

Comment:
If you go with different Head interfaces, can this line be changed to the following, and the 1st generic can then be removed:
HTH htrktypes.Head[commontypes.Head[BLOCK_HASH], BLOCK_HASH, ID],

Reply:
that's much cleaner! thanks for the note

Reply:
That actually goes back to the `Equals()` method where because we want `Equals(r *Head)` as a method for `Head` instead of `Equals(r commontypes.Head[common.Hash])`. 

Therefore if we do `HTH htrktypes.Head[commontypes.Head[BLOCK_HASH], BLOCK_HASH, ID]`, instead of the pointer receiver, it would require `Equals(r commontypes.Head[common.Hash])`.

Reply:
it does seem a little odd to have the head types repeated...if it hinges on the `Equals` func could we just pass in a comparison func to the struct that is specific to EVM and then just use a single head type?

that way we might be able to separate out the part that is causing some strange syntax

similar to how the txm passes in a `isNil` func? https://github.com/smartcontractkit/chainlink/blob/c8f144bc05d27e75850ed3d9c982acf14dbfd8bf/core/chains/evm/txmgr/confirmer.go#L143

Reply:
thanks for suggesting this approach. I think thats a great approach to remove clutter for Equals. Will be removing the interface for `HeadTrackerHead` and condensing into 1 interface.

Reply:
on top of the `isNil`, another aspect would be around avoiding propagating changes because of the need to define another generic for the HEAD - `ChainID`